diff --git a/interop/src/interop.cpp b/interop/src/interop.cpp
index 4e3e14e..f837269 100644
--- a/interop/src/interop.cpp
+++ b/interop/src/interop.cpp
@@ -1,7 +1,7 @@
 ﻿#include "interop.hpp"
 
-#include "../external/Spout2/SPOUTSDK/SpoutGL/SpoutReceiver.h"
-#include "../external/Spout2/SPOUTSDK/SpoutGL/SpoutSender.h"
+#include "SpoutGL/SpoutReceiver.h"
+#include "SpoutGL/SpoutSender.h"
 //#include "SpoutReceiver.h"
 //#include "SpoutSender.h"
 #include <nlohmann/json.hpp>
