diff --git a/CMakeLists.txt b/CMakeLists.txt
index 6e52463..26d1e6c 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -21,8 +21,9 @@ if (MSF_INTEGRATED_GLSLANG)
   include(cmake/fetch_glslang.cmake)
 else ()
   # vcpkg support
+  find_package(Threads REQUIRED)
   find_package(glslang REQUIRED)
-  target_include_directories(glslang-default-resource-limits INTERFACE
+  target_include_directories(glslang::glslang-default-resource-limits INTERFACE
     $<BUILD_INTERFACE:${CMAKE_CURRENT_SOURCE_DIR}/cmake/glslang_workaround>)
 endif ()
 
@@ -40,7 +41,7 @@ target_compile_features(msf_static PUBLIC cxx_std_17)
 set_target_properties(msf_static PROPERTIES
   CXX_EXTENSIONS OFF
   POSITION_INDEPENDENT_CODE ON)
-target_link_libraries(msf_static PRIVATE glslang glslang-default-resource-limits)
+target_link_libraries(msf_static PRIVATE glslang::glslang glslang::glslang-default-resource-limits)
 target_include_directories(msf_static
   PUBLIC
     $<INSTALL_INTERFACE:include>
