diff --git a/third_party/CMakeLists.txt b/third_party/CMakeLists.txt
index cd68b1f..a791a96 100644
--- a/third_party/CMakeLists.txt
+++ b/third_party/CMakeLists.txt
@@ -39,6 +39,7 @@ endif ()
 
 
 # cURL
+if(0)
 FetchContent_Declare(curl
     URL "https://github.com/curl/curl/archive/refs/tags/curl-8_2_1.zip"
 )
@@ -93,7 +94,7 @@ mark_as_advanced(FORCE
     USE_QUICHE
     USE_WIN32_IDN
     USE_WIN32_LDAP)
-
+endif()
 
 # JSON library
 FetchContent_Declare(nlohmann_json
